Core/src/Detector/detail/SupportHelper.cpp

```diff
@@ -215,9 +236,50 @@ void Acts::Experimental::detail::SupportHelper::addSupport(
       layerSurfaces.insert(layerSurfaces.end(), dSupport.begin(),
                            dSupport.end());
     }
+  } else if (layerRepresentation == Surface::SurfaceType::Plane) {
+    if (not layerExtent.constrains(binX) or not layerExtent.constrains(binY) or
+        not layerExtent.constrains(binZ)) {
+      throw std::runtime_error(
+          "SupportHelper::addSupport(...) - x,y or z are not constrained, "
+          "needed for rectangular support.");
+    }
+
+    // Get the main support parameters:
+    // - doff .. offset (in r.z)
+    // - dl0min, dl0max .. envelop min, max (in z,r)
+    // - dl1min, dl1max .. envelop min, max (in phi)
+    auto [doff, dl0min, dl0max, dl1min, dl1max] = layerSupportValues;
```

Comment:
I guess this, and the calculations below, assumes that the envelopes are all positive numbers. Is this check performed somewhere, or also negative values are possible?

Reply:
I think I will stick with garbage in garbage out ... but we can add a exception check.